src/sdl3/video.rs

```diff
@@ -2281,6 +2330,40 @@ impl Window {
             Err(get_error())
         }
     }
+
+    /// Sets a hit test function for the window.
+    #[doc(alias = "SDL_SetWindowHitTest")]
+    pub fn set_hit_test(
+        &mut self,
+        hit_test: impl Fn(crate::rect::Point) -> HitTestResult,
+    ) -> Result<(), Error> {
+        // Box the closure to extend its lifetime and convert it to a raw pointer.
+        let boxed: Box<Box<dyn Fn(crate::rect::Point) -> HitTestResult>> =
+            Box::new(Box::new(hit_test));
+        let userdata = Box::into_raw(boxed) as *mut c_void;
+
+        unsafe extern "C" fn hit_test_sys(
+            _: *mut sys::video::SDL_Window,
+            point: *const sys::rect::SDL_Point,
+            data: *mut c_void,
+        ) -> sys::video::SDL_HitTestResult {
+            // Reborrow the boxed closure.
+            let callback = data as *mut Box<dyn Fn(crate::rect::Point) -> HitTestResult>;
+            let point = crate::rect::Point::from_ll(*point);
+
+            (*callback)(point).to_ll()
+        }
+
+        unsafe {
+            let result =
+                sys::video::SDL_SetWindowHitTest(self.context.raw, Some(hit_test_sys), userdata);
```

Comment:
It seems like the hit_test will never be freed, we could store the userdata pointer on the Window struct and free it on Window drop perhaps.